client/quantum_serverless/core/job.py

```diff
@@ -375,7 +375,11 @@ def result(self, wait=True, cadence=5, verbose=False):
                 time.sleep(cadence)
                 if verbose:
                     logging.info(".")
-        return self._job_client.result(self.job_id)
+        results = self._job_client.result(self.job_id)
+        if isinstance(results, str):
```

Comment:
I could be overthinking things here, but would it be better to handle this within the job client (i.e. [here](https://github.com/Qiskit-Extensions/quantum-serverless/blob/c2a5e9d6d5adde09eb20590e9163298ada593154/client/quantum_serverless/core/job.py#L286))?

Reply:
It looks like the return is already loaded into a dictionary in that line, but when I use `save_result` inside my remote program, `job.result()` returns me a string representation of a dictionary

Reply:
OK I see what's happening. I think I am intended to use the client to retrieve results, but I was using the `job` object itself. I guess it's appropriate to do the decoding on both levels?

Reply:
@IceKhan13 why would the return object in the deleted line ever be a string? `job_client.result()` returns a decoded object [here](https://github.com/Qiskit-Extensions/quantum-serverless/blob/c2a5e9d6d5adde09eb20590e9163298ada593154/client/quantum_serverless/core/job.py#L286)

Reply:
Hmm, good question. It should be decoded object (dict). I do not know why it's returning string. Worth digging into it